lib/mapping/MapFormatH3M.cpp

```diff
@@ -1842,77 +1854,89 @@ void CMapLoaderH3M::readSeerHutQuest(CGSeerHut * hut, const int3 & position)
 
 	if(hut->quest->missionType)
 	{
-		auto rewardType = static_cast<CGSeerHut::ERewardType>(reader->readUInt8());
-		hut->rewardType = rewardType;
+		auto rewardType = reader->readUInt8();
+		Rewardable::VisitInfo vinfo;
+		auto & reward = vinfo.reward;
 		switch(rewardType)
 		{
-			case CGSeerHut::EXPERIENCE:
+			case 0: //NOTHING

```

Comment:
Please keep the enum for sake of readability. You can move it to `CRewardableObject` or `MapFormaH3M` class.

Reply:
Yes. Should be an enum in `MapFormatH3M`, similar to `EVictoryConditionType`